src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -27,508 +27,303 @@ public class ResultTextMarker
+                // otherwise users cannot move the caret in the region.
+                if (trackingSpanSnapshot.Contains(_wpfTextView.Caret.Position.BufferPosition))
                 {
-                    System.Diagnostics.Process.Start(uri.OriginalString);
-                    return null;
+                    _vsWindowFrame?.Show();
+                    return true;
                 }
-            }
 
-            // Fall back to the file and line number
-            if (!File.Exists(this.FullFilePath))
-            {
-                if (!CodeAnalysisResultManager.Instance.TryRebaselineAllSarifErrors(_runId, this.UriBaseId, this.FullFilePath))
+                // The caret is not in this result, move it there so the result can be seen.
+                if (!trackingSpanSnapshot.IsEmpty)
                 {
-                    return null;
+                    _wpfTextView.Selection.Select(trackingSpanSnapshot, isReversed: false);
+                    _wpfTextView.Caret.MoveTo(trackingSpanSnapshot.End);
+                    _wpfTextView.Caret.EnsureVisible();
+                    _vsWindowFrame?.Show();
                 }
-            }
-            
-            if (File.Exists(this.FullFilePath) && Uri.TryCreate(this.FullFilePath, UriKind.Absolute, out uri))
-            {
-                // Fill out the region's properties
-                FileRegionsCache regionsCache = CodeAnalysisResultManager.Instance.RunDataCaches[_runId].FileRegionsCache;
-                Region = regionsCache.PopulateTextRegionProperties(Region, uri, true);
+
+                return true;
             }
 
-            IVsWindowFrame windowFrame = SdkUIUtilities.OpenDocument(ServiceProvider.GlobalProvider, this.FullFilePath, usePreviewPane);
-            if (windowFrame != null)
+            if (retryNaviation)
             {
-                IVsTextView textView = GetTextViewFromFrame(windowFrame);
-                if (textView == null)
+                // If we get here, this marker hasn't yet been attached to a document and therefore
+                // will attempt to open the document and select the appropriate line.
+                if (!File.Exists(this.FullFilePath))
                 {
-                    return null;
+                    if (!CodeAnalysisResultManager.Instance.TryRebaselineAllSarifErrors(_runIndex, this.UriBaseId, this.FullFilePath))
+                    {
+                        return false;
+                    }
                 }
 
-                var sourceLocation = this.GetSourceLocation();
+                if (File.Exists(this.FullFilePath) && Uri.TryCreate(this.FullFilePath, UriKind.Absolute, out Uri uri))
+                {
+                    // Fill out the region's properties
+                    FileRegionsCache regionsCache = CodeAnalysisResultManager.Instance.RunIndexToRunDataCache[_runIndex].FileRegionsCache;
+                    Region = regionsCache.PopulateTextRegionProperties(Region, uri, true);
```

Comment:
>PopulateTextRegionProperties [](start = 42, length = 28)

Confirmed. After this call, all of `region.startLine`, `.endLine`, `.startColumn`, `.endColumn`, `.charOffset`, and `.charLength` are guaranteed to be populated.